### DISCUSSION TITLE
Unable to access docker registry ENV variables

### DISCUSSION THREAD

**Discussion by mego22** [community/community#6124] [Codespaces]
I have the three ENV variables set-GHCR_CONTAINER_REGISTRY_SERVER, GHCR_CONTAINER_REGISTRY_USER, and GHCR_CONTAINER_REGISTRY_PASSWORD-to access ghcr.io but when I start up a Codespace they are not set. How do I access these secrets?

**Comment by asciimike** [comment]
@mego22 to confirm, you've implemented these as secrets as suggested [in the docs](https://docs.github.com/en/codespaces/codespaces-reference/allowing-your-codespace-to-access-a-private-image-registry)? What do you get when you run `docker login` or `docker pull`?

**Comment by metaskills** [comment]
From testing, it is my understanding that the secrets following this naming convention are not exposed to users but are used lower down when Codespaces builds your development environment.
